polar-core/src/parser.rs

```diff
@@ -240,6 +240,22 @@ mod tests {
         assert_eq!(results[0].to_polar(), r#"a(x) if x = new Foo(1, 2);"#);
     }
 
+    #[test]
+    fn test_parse_new_mixed_args() {
+        let f = r#"a(x) if x = new Foo(1, 2, bar: 3);"#;
+        let results = parse_rules(0, f).unwrap();
+        assert_eq!(
+            results[0].to_polar(),
+            r#"a(x) if x = new Foo(1, 2, bar: 3);"#
+        );
+        let f = r#"a(x) if x = new Foo(bar: 3, baz: 4);"#;
+        let results = parse_rules(0, f).unwrap();
+        assert_eq!(
+            results[0].to_polar(),
+            r#"a(x) if x = new Foo(bar: 3, baz: 4);"#
+        );
+    }
```

Comment:
nit: It would be great to add a case that expects an error if a positional arg trails a kwarg